src/System.Text.JsonLab.Serialization/System/Text/Json/Serialization/JsonClassInfo.cs

```diff
@@ -193,12 +134,39 @@ public JsonPropertyInfo GetProperty(Type classType, ReadOnlySpan<byte> propertyN
             return info;
         }
 
+        public JsonPropertyInfo GetPolicyProperty()
+        {
+            Debug.Assert(_property_refs.Count == 1);
+            return _property_refs[0].Info;
+        }
+
+        public JsonPropertyInfo GetProperty(int index)
+        {
+            Debug.Assert(index < _property_refs.Count);
+            return _property_refs[index].Info;
+        }
+
+        public ReadOnlySpan<byte> GetPropertyName(int index)
+        {
+            Debug.Assert(index < _property_refs.Count);
+            return _property_refs[index].Info.Name;
+        }
+
+        public int PropertyCount
+        {
+            get
+            {
+                return _property_refs.Count;
+            }
+        }
+
         private static bool TryIsPropertyRefEqual(List<PropertyRef> list, ReadOnlySpan<byte> propertyName, ulong key, int index, out JsonPropertyInfo info)
         {
             if (key == list[index].Key)
             {
                 if (propertyName.Length <= PropertyNameKeyLength ||
-                    propertyName.Slice(PropertyNameKeyLength).SequenceEqual((ReadOnlySpan<byte>)list[index].Info.Key))
+                    // todo: is it really any faster to do slices here to avoid comparing the first 6 bytes?
+                    propertyName.Slice(PropertyNameKeyLength).SequenceEqual(((ReadOnlySpan<byte>)list[index].Info.Name).Slice(PropertyNameKeyLength)))
```

Comment:
Slicing spans is generally quite cheap, but something we should measure, especially taking into account common property name lengths.